parser.js

```diff
@@ -542,14 +540,15 @@ class Parser extends EventEmitter {
 
   _parseVarByteNum (fullInfoFlag) {
     debug('_parseVarByteNum')
+    const maxBytes = 4
     let bytes = 0
     let mul = 1
     let value = 0
     let result = false
     let current
     const padding = this._pos ? this._pos : 0
 
-    while (bytes < 5) {
+    while (bytes < maxBytes) {
```

Comment:
shouldn't this be a less than or equal for maxBytes

Reply:
I believe that the previous ' bytes < 5' was incorrect.

The maximum size of a var byte int is 4 bytes, so 4 bytes need to be examined in the loop.
The bytes variable starts at 0, so to examine 4 bytes, the loop has to be over 0, 1, 2, 3.
If it doesn't meet the ((current & constants.VARBYTEINT_FIN_MASK) === 0) condition for any of those four bytes, then that indicates that it has exceeded the maximum size, i.e. that bit being non-zero means that it is expecting more length bytes, but we are already at the maximum at for bytes === 3.

Here are a couple of tests that demonstrate that buggy aspect present in v6.6.0, failing to emit the required error there:
```
testParseError('Invalid length', Buffer.from(
  [16, 255, 255, 255, 255, 1]
), {})
testParseError('Invalid length', Buffer.from(
  [16, 255, 255, 255, 255, 127]
), {})
```

While the following tests do pass in v6.6.0:
```
testParseError('Invalid length', Buffer.from(
  [16, 255, 255, 255, 255, 128]
), {})
testParseError('Invalid length', Buffer.from(
  [16, 255, 255, 255, 255, 255, 1]
), {})
```

(Note that the error message is different between v6.6.0 and my changes, i.e. 'Invalid length' vs 'Invalid variable byte integer'.)

With the updated error message, the following tests all pass with my changes:

```
testParseError('Invalid variable byte integer', Buffer.from(
  [16, 255, 255, 255, 255]
), {})
testParseError('Invalid variable byte integer', Buffer.from(
  [16, 255, 255, 255, 128]
), {})
testParseError('Invalid variable byte integer', Buffer.from(
  [16, 255, 255, 255, 255, 1]
), {})
testParseError('Invalid variable byte integer', Buffer.from(
  [16, 255, 255, 255, 255, 127]
), {})
testParseError('Invalid variable byte integer', Buffer.from(
  [16, 255, 255, 255, 255, 128]
), {})
testParseError('Invalid variable byte integer', Buffer.from(
  [16, 255, 255, 255, 255, 255, 1]
), {})
```

That seems to be good evidence that my version has the correct while condition and that the previous one was another bug in v6.6.0.